packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -641,7 +641,8 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         existing?: boolean,
     ): Promise<ContainerRuntime> {
         // If taggedLogger exists, use it. Otherwise, wrap the vanilla logger:
-        const passLogger = context.taggedLogger  ?? new TaggedLoggerAdapter(context.logger);
+        // back-compat: Remove the TaggedLoggerAdapter fallback once all the host are using loader > 0.45
+        const passLogger = context.taggedLogger  ?? new TaggedLoggerAdapter((context as any).logger);
```

Comment:
_Maybe this idea is overkill, you can decide, but it's more explicit about the compat concerns here and avoids `as any` which always helps me sleep better at night..._

I'm wondering about defining a type (not exported) in here like this:

```
interface OldContainerContextWithLogger extends IContainerContext { logger: ITelemetryBaseLogger }
```

Then do `context as OldContainerContextWithLogger` instead of `as any`.

And you can put a nice comment on that interface definitions explaining that while an untagged logger is unsupported moving forward, in the wild there are old loaders with old ContainerContexts that only have the untagged logger, so we need to accommodate that scenario.

Reply:
@markfields I was thinking about this same thing on the other PR. I was conflicted there too, and then I thought if I really had more than one occurance where I could refer to SomeOldInterface, I would probably go down that route. It would also help ensure typing, correlate changes to eachother, and make search/removal easier. But since I had one-liner (like this) change that is commented with back-compat template, I left it as is. But yes - I like the idea!

Reply:
that's a great idea!! do you think the interface should also be deprecated with a comment saying "it can be removed once its usage is removed from `TaggedLoggerAdapter` fallback"